### PR TITLE
 Set storage types for reader cluster

### PR DESCRIPTION
**Pull request by lcadzy**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageWithUploadPathPanel.kt [hunk 1]
```diff
@@ -168,6 +171,29 @@ class SparkSubmissionJobUploadStorageWithUploadPathPanel
...
         }
 
+        private fun setUploadInfoForLinkedCluster(cluster: IClusterDetail, model: SparkSubmitJobUploadStorageModel) {
+            model.apply {
+                val defaultFS = (cluster as? HDInsightAdditionalClusterDetail)?.defaultFS
+                defaultFS?.run {
+                    // get account and container for blob or adls gen2
+                    var fullAccountName: String? = null
+                    var container: String? = null
+                    val m = Pattern.compile(ClusterDetail.StorageAccountNamePattern).matcher(defaultFS)
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageWithUploadPathPanel.kt]: Is this mather only for Blob?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterDetail.java [hunk 2]
```diff
@@ -53,7 +53,7 @@
     private final String DefaultFS = "fs.defaultFS";
     private final String FSDefaultName = "fs.default.name";
     private final String StorageAccountKeyPrefix = "fs.azure.account.key.";
-    private final String StorageAccountNamePattern = "^(abfs[s]?|wasb[s]?)://(.*)@(.*)$";
+    public static final String StorageAccountNamePattern = "^(?<scheme>abfs[s]?|wasb[s]?)://(?<container>.*)@(?<fullAccountName>.*)$";
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterDetail.java]: This one can be moved to StorageAccount class

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/HDInsightAdditionalClusterDetail.java [hunk 3]
```diff
@@ -113,18 +122,49 @@ public String getHttpPassword() {
...
+        if (StringUtils.isNullOrEmpty(defaultFS)) {
+            return SparkSubmitStorageTypeOptionsForCluster.HdiAdditionalClusterWithUndetermineStorage;
+        }
+
+        // blob or adls gen2 storage account
+        if (defaultFS.matches(ClusterDetail.StorageAccountNamePattern)) {
+            Matcher m = Pattern.compile(ClusterDetail.StorageAccountNamePattern).matcher(defaultFS);
+            String scheme = null;
+            if (m.find()) {
+                scheme = m.group("scheme");
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/HDInsightAdditionalClusterDetail.java]: Can this parsing code be combined into one class?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java [hunk 4]
```diff
@@ -102,6 +111,29 @@
         this.optionTypes = optionTypes;
     }
 
+    SparkSubmitStorageTypeOptionsForCluster(StorageAccountTypeEnum type) {
+        switch(type){
+            case BLOB:
+            this.optionTypes = new SparkSubmitStorageType[]{
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java]: coding style issue

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java [hunk 5]
```diff
@@ -26,6 +26,8 @@
  */
 package com.microsoft.azure.hdinsight.spark.common;
 
+import com.microsoft.azure.hdinsight.sdk.storage.StorageAccountTypeEnum;
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java]: Could you help to rename the class name without type ending?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java (hunk 4 above)
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitStorageTypeOptionsForCluster.java]: A tip in IntelliJ: We can select a piece of codes and apply `Ctrl + Alt + L` to auto-format the codes.

---
In reply to: [277987651](https://github.com/Microsoft/azure-tools-for-java/pull/3074#discussion_r277987651) [](ancestors = 277987651)

**Pull request merged**